### PR TITLE
Export `QueryForPathname`

### PR DESCRIPTION
**Pull request by MariaSolOs**
(I promise this is the last PR I create today lol)

Just adding this type export so that one can easily type queries in NextJS data fetching functions (e.g.: `GetStaticProps`).

The use case I'm considering is something like:
```
// In blogs/[id].tsx

import type { GetStaticProps } from 'next';
import type { QueryForPathname } from 'next/router';

export const getStaticProps: GetStaticProps<{}, QueryForPathname['/blogs/[id]']> = async (context) => {
    const { id } = context.params; // This will be typed! :)
}
```

### CODE REVIEW

**Comment by tatethurston** [comment]
Awesome, thanks for the contribution and outlining the use case.

**Comment by tatethurston** [comment]
We'll want to export the type from the block

```
declare module "nextjs-routes" {
```

instead of from:

```
declare module "next/router" {
```

And add type tests to `e2e/typetest.tsx`.

What do you think about naming this `Query`? Something like:

```ts
type Query<Pathname extends Route["pathname"] = Route["pathname"]> = Extract<
  Route,
  { pathname: Pathname }
>['query'];
```

**Comment by MariaSolOs** [comment]
@tatethurston Sounds good, I'll move the export and update the tests.

The `nextjs-routes` module already has a type named `Query`. What about `RoutedQuery`? That makes it clear that this is the typed query object of a specific route.

**Comment by tatethurston** [comment]
Yeah that sounds great

**Comment by MariaSolOs** [comment]
@tatethurston QQ: How am I supposed to update the `nextjs-routes.d.ts` file in the `e2e` package? It seems like it has been previously generated by the CLI but isn't that getting deprecated?

**Comment by tatethurston** [comment]
Just mergeed https://github.com/tatethurston/nextjs-routes/pull/70, you can `npm test` to regenerate this file.

Alternatively `next build` or `next dev` in the e2e directory will regenerate the file.

File: src/__snapshots__/core.test.ts.snap [hunk 1]
```diff
@@ -15,6 +15,10 @@ declare module "nextjs-routes" {
 
   type Query<Params = {}> = Params & { [key: string]: string | string[] | undefined };
 
+  export type RoutedQuery = {
+    [K in Route as K["pathname"]]: Exclude<K["query"], undefined>;
+  };
```
Comment (tatethurston) [inline src/__snapshots__/core.test.ts.snap:20]:
```suggestion
  type QueryForPathname = {
    [K in Route as K["pathname"]]: Exclude<K["query"], undefined>;
  };
  
  export type RoutedQuery<Pathname extends Route["pathname"] = Route["pathname"]> = QueryForPathname[Pathname]
```
Could you export a generic type instead? That way the syntax is consistent for end users:

```
RoutedQuery<"/foos/[foo]">
```
vs
```
RoutedQuery["/foos/[foo]"]
```

**Review by tatethurston — APPROVED** [review]

**Pull request merged**